oci/layout/oci_dest.go

```diff
@@ -167,11 +171,19 @@ func (d *ociImageDestination) PutManifest(m []byte) error {
 	if err := ioutil.WriteFile(d.ref.ociLayoutPath(), []byte(`{"imageLayoutVersion": "1.0.0"}`), 0644); err != nil {
 		return err
 	}
-	descriptorPath := d.ref.descriptorPath(d.ref.tag)
-	if err := ensureParentDirectoryExists(descriptorPath); err != nil {
-		return err
-	}
-	return ioutil.WriteFile(descriptorPath, data, 0644)
+
+	annotations := make(map[string]string)
+	annotations["org.opencontainers.ref.name"] = d.ref.tag
+	desc.Annotations = annotations
+	d.index.Manifests = append(d.index.Manifests, imgspecv1.ManifestDescriptor{
```

Comment:
Ah, true. I wasn’t considering that because we have no tools to do that (and the semantics of doing that + `PutSignatures` is unclear — what manifest are the signatures attached to?).

Anyway, that can be useful for someone who explicitly expects that `oci:` behavior, and it doesn’t break anything, so fair enough.